src/lib/pickles/intf.mli

```diff
@@ -161,40 +56,61 @@ module Group (Impl : Snarky_backendless.Snark_intf.Run) : sig
 
       val scale : t -> Scalar.t -> t
 
+      (** [to_affine_exn p] returns the affine coordinates [(x, y)] of the point
+          [p] *)
       val to_affine_exn : t -> Impl.field * Impl.field
 
+      (** [of_affine (x, y)] builds a point on the curve
+          TODO: check it is on the curve? Check it is in the prime subgroup?
+      *)
       val of_affine : Impl.field * Impl.field -> t
     end
 
+    (** Represent a point, but not necessarily on the curve and in the prime
+        subgroup *)
     val typ_unchecked : (t, Constant.t, Impl.field) Snarky_backendless.Typ.t
 
+    (** Represent a point on the curve and in the prime subgroup *)
     val typ : (t, Constant.t, Impl.field) Snarky_backendless.Typ.t
 
+    (** Add two points on the curve.
+        TODO: is the addition complete?
+    *)
     val ( + ) : t -> t -> t
 
+    (** Double the point *)
     val double : t -> t
 
+    (** [scalar g xs] computes the MSM of [g] with [xs] *)
```

Comment:
Is this an MSM OR just a single scalar multiplication where the scalar is passed expanded already into a list of bits?